packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,453 @@
+     */
+    function accessoriesPageCount() public view override returns (uint256) {
+        return _accessories.storagePages.length;
+    }
+
+    /**
+     * @notice Get the number of glasses storage pages.
+     */
+    function glassesPageCount() public view override returns (uint256) {
+        return _glasses.storagePages.length;
+    }
+
+    /**
+     * @notice Get a storage page of head images.
+     */
+    function headsPage(uint256 pageIndex) public view override returns (INounsArt.NounArtStoragePage memory) {
+        return _heads.storagePages[pageIndex];
+    }
+
+    /**
+     * @notice Get a storage page of body images.
+     */
+    function bodiesPage(uint256 pageIndex) public view override returns (INounsArt.NounArtStoragePage memory) {
+        return _bodies.storagePages[pageIndex];
+    }
+
+    /**
+     * @notice Get a storage page of accessory images.
+     */
+    function accessoriesPage(uint256 pageIndex) public view override returns (INounsArt.NounArtStoragePage memory) {
+        return _accessories.storagePages[pageIndex];
+    }
+
+    /**
+     * @notice Get a storage page of glasses images.
+     */
+    function glassesPage(uint256 pageIndex) public view override returns (INounsArt.NounArtStoragePage memory) {
+        return _glasses.storagePages[pageIndex];
+    }
+
+    /**
+     * @notice Get a head image bytes (RLE-encoded).
+     */
+    function heads(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_heads, index);
+    }
+
+    /**
+     * @notice Get a body image bytes (RLE-encoded).
+     */
+    function bodies(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_bodies, index);
+    }
+
+    /**
+     * @notice Get a accessory image bytes (RLE-encoded).
+     */
+    function accessories(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_accessories, index);
+    }
+
+    /**
+     * @notice Get a glasses image bytes (RLE-encoded).
+     */
+    function glasses(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(_glasses, index);
+    }
+
+    /**
+     * @notice Get a color palette bytes.
+     */
+    function palettes(uint8 paletteIndex) public view override returns (bytes memory) {
+        return SSTORE2.read(palettesPointers[paletteIndex]);
+    }
+
+    function _addBackground(string calldata _background) internal {
+        backgrounds.push(_background);
+    }
+
+    function addToTraitFromBytes(
+        Trait storage trait,
+        bytes calldata encodedCompressed,
+        uint80 decompressedLength,
+        uint16 imageCount
+    ) internal {
+        if (encodedCompressed.length == 0) {
+            revert EmptyBytes();
+        }
+        address pointer = SSTORE2.write(encodedCompressed);
+        addToTraitFromPointer(trait, pointer, decompressedLength, imageCount);
+    }
+
+    function addToTraitFromPointer(
```

Comment:
nit: rename to `addPage` ? the different parameter types express the difference from the function above